### ISSUE TITLE
Do we want to move IUrlHelper and friends?

### ISSUE THREAD

**Issue by rynowak** [aspnet/Mvc#3677]
We had earlier tried to move the namespace of `IUrlHelper` and I believe a few others that are referenced here: https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorHost.cs#L41

The issue is that these namespaces are hardcoded into tooling. If we move these, then intellisense break for these properties.

I don't recall if we ever resolved this, but it came up during discussion related to #3652

**Comment by danroth27** [comment]
What is the proposal for the new namespaces? Is the goal simply to clean up the top level MVC namespace?

If we did decide to do this could we just fix the tooling?

**Comment by rynowak** [comment]
The propose is that we change the names to be what we decided they should be when we did the API review. This was part of cleaning up the top-level namespaces that we punted on in beta8.

> If we did decide to do this could we just fix the tooling?

Making this change would fix the tooling. It would just break compatibility across rc1-rc2 in a somewhat grievous way. This is because tooling ships a build of Razor.
